ztrouy/gondor-backend
# Complete Initial Seed Data

### Purpose
To create seed data for all custom models, and fix seed data and models that did not quite meet project requirements. For Appointments, Patient Data, Reschedule Requests, and Logs, the seed data is generated when seeding the database. This ensures that there are always actionable tasks for each Group.

### Files Changed
- Added a "user" property to link Addresses to the User that added them in `address.py` model
- Updated "approver" and "approved_timestamp" properties to be nullable in `appointment.py` model
- Updated the "updated_by" and "updated_timestamp" properties to be nullable in `patient_data.py` model
- Added an "approved_timestamp" property, and updated the "approver" property to be nullable in `reschedule_request.py` model
- Added seed data for new "user" property to all Addresses, and added new Addresses in `addresses.json` fixture
- Removed "_id" from each property in `user_addresses.json` fixture
- Created package files for new commands in new `__init__.py` packages in `/gondorapi/management` and `/gondorapi/management/commands` directories
- Created scripts callable as commands that generate up-to-date seed data for Appointments, Reschedule Requests, Patient Data, and Logs in `seed_appointments.py`, `seed_reschedule_requests.py`, `seed_patient_data.py`m and `seed_logs.py` respectively
- Added commands for generating Appointments, Reschedule Requests, Patient Data, and Logs seed data in `seed_database.sh`

### To Test
Fetch, then run the following commands in the project root directory
```
pipenv shell
```
```
pipenv install
```
```
./seed_database.sh
```
Then confirm no errors occur during the process of the final command. Rerun final command multiple times to ensure an error never occurs.

closes #8
